File: backend/src/main/java/com/campusshare/entity/UserReport.java
package com.campusshare.entity;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class UserReport {
    private Long id;

    private Long reportedUserId;

    private Long userId;

    private String reason;

    private String status;

    private String reviewReply;

    private String reportedUsername;

    private Boolean cancelled;

    private LocalDateTime createdAt;

    private LocalDateTime reviewedAt;

    private LocalDateTime updatedAt;
}

File: frontend/src/api/userReport.ts
import request from './request'

export interface UserReport {
  id: number
  reportedUserId: number
  reportedUsername?: string
  userId: number
  reason: string
  status: string
  reviewReply?: string
  cancelled: boolean
  createdAt: string
  reviewedAt?: string
  updatedAt?: string
}

export function createUserReport(reportedUserId: number, reason: string) {
  return request.post<{ success: boolean; data: UserReport }>('/user-reports', {
    reportedUserId,
    reason,
  })
}

export function getMyUserReports(status?: string) {
  return request.get<{ success: boolean; data: UserReport[] }>('/user-reports/my', {
    params: status ? { status } : {},
  })
}
